### ISSUE TITLE
Missing .debs for 0.3.3?

### ISSUE THREAD

**Issue by anowlcalledjosh** [jwilm/alacritty#2554]
Version 0.3.2 has deb packages available in its GitHub releases and in the [mmstick76 PPA](https://launchpad.net/~mmstick76/+archive/ubuntu/alacritty), but there are no debs for 0.3.3 or any of its RCs. I couldn't find any discussion of this on the issue tracker, or any related commits. Is this intended?

**Comment by chrisduerr** [comment]
I haven't noticed this since the build for the .deb did not fail, however you are correct and it is not intentional.

I'm working on getting the deb build up and I'll send a separate PR to make error reporting a bit more strict.

**Comment by rsimmonsjr** [comment]
I am eager for the fixes, cant wait.

**Comment by chrisduerr** [comment]
I've uploaded manually compiled .deb files to the release for now, I'll close this issue once my PR to fix this permanently is merged.

Let me know if there are any issues.

**Comment by rsimmonsjr** [comment]
When will they make it to the apt repository ?

**Comment by chrisduerr** [comment]
If you're talking about the popos ppa, that's up to the maintainer @mmstick. Chances are it's just a matter of time. I'd recommend just giving them a few days and chances are they'll have it all up and running. Depending on testing procedures and number of packages per maintainer, response times can obviously vary a bit.

**Comment by mmstick** [comment]
I will have it updated today. I check for package updates in my feed every Monday.

**Comment by chrisduerr** [comment]
Thanks for the info. For now the .deb build should work automatically again too, though with the PPA it's not unlikely that Alacritty will stop providing debian builds (since the PPA is a lot higher quality and has better platform support).

Though for the next few releases both the PPA and the .deb/.tar.gz should be available.

**Comment by anowlcalledjosh** [comment]
FYI, the postinst and prerm scripts are missing from the 0.3.3 deb – it's not a huge deal, since it's just running `update-alternatives`, but I thought I'd mention it.

**Comment by chrisduerr** [comment]
@anowlcalledjosh Do you have Rust installed and are you capable of building the .deb yourself? I'd be curious if the maintainer scripts work there.

If not, I'd expect that [this line](https://github.com/jwilm/alacritty/blob/master/alacritty/Cargo.toml#L57) would probably need to have the `..` removed. So if it doesn't work for you following the `INSTALL.md` instructions, testing that would be much appreciated.

**Comment by anowlcalledjosh** [comment]
The maintainer scripts are not included in a deb built from the `v0.3.3` tag:

```console
$ dpkg -I target/debian/alacritty_0.3.3_amd64.deb
 new Debian package, version 2.0.
 size 5016510 bytes: control archive=801 bytes.
     697 bytes,    14 lines      control
     562 bytes,     8 lines      md5sums
 Package: alacritty
```

Removing the `../` as suggested results in them being included:

```console
$ dpkg -I target/debian/alacritty_0.3.3_amd64.deb
 new Debian package, version 2.0.
 size 5016564 bytes: control archive=912 bytes.
     697 bytes,    14 lines      control
     562 bytes,     8 lines      md5sums
     115 bytes,     3 lines   *  postinst             #!/bin/sh
      82 bytes,     3 lines   *  prerm                #!/bin/sh
 Package: alacritty
```

I noticed that the files in `assets` are still apparently included even though they have the `../` prefix, though, and `cargo deb` exits with an error if that prefix is removed, which seems odd. I won't pretend I understand Cargo workspaces, but this seems like a bug or at least a wart in cargo-deb, and applying the following patch to it gets the maintainer scripts to be included in alacritty's deb with no changes to Cargo.toml:

```diff
diff --git a/src/control.rs b/src/control.rs
index 1b013f0cf..8435e14d1 100644
--- a/src/control.rs
+++ b/src/control.rs
@@ -26,7 +26,7 @@ pub fn generate_archive(options: &Config, time: u64, asset_hashes: HashMap<PathB
 fn generate_scripts(archive: &mut Archive, option: &Config) -> CDResult<()> {
     if let Some(ref maintainer_scripts) = option.maintainer_scripts {
         for name in &["config", "preinst", "postinst", "prerm", "postrm"] {
-            if let Ok(script) = fs::read(maintainer_scripts.join(name)) {
+            if let Ok(script) = fs::read(option.path_in_workspace(maintainer_scripts).join(name)) {
                 archive.file(name, &script, 0o755)?;
             }
         }
```

If that seems like the correct behaviour I'd be happy to submit that as a PR to cargo-deb. (Or I'd also be happy to be told I'm misunderstanding and that cargo-deb's current behaviour is correct!)

**Comment by praseodym** [comment]
Just ran across the same issue (postinst script not being included in the deb package). Should I open a new issue to track this?
